README.md

```diff
@@ -1,33 +1,63 @@
+# Installing Build Dependencies
 
-# Build the cross-compiler
+The [main build script](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/build.sh) installs its dependencies automatically when `RUN_BOOTSTRAP=1` environment variable is defined which is the default. To see what will be installed refer to [`.github/scripts/install-dependencies.sh`](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/.github/scripts/install-dependencies.sh).
 
-Running build.sh will build the cross compiler. It installs it into ~/cross. The build script does multiple things.
- - Downloads dependencies
- - Clones several dependencies 
-    - modified binutils from https://github.com/Windows-on-ARM-Experiments/binutils-woarm64.git
-    - modified gcc from https://github.com/Windows-on-ARM-Experiments/gcc-woarm64
-    - modified mingw from https://github.com/Windows-on-ARM-Experiments/mingw-woarm64
- - Builds and installs: binutils, mingw, gcc and libgcc.
+# Building the Cross-Compiler
 
-# Using the cross compiler
+Running [`./build.sh`](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/build.sh) on Ubuntu 22.04 in the root repository folder will build the cross-compiler and install it into `~/cross-aarch64-w64-mingw32-msvcrt` folder by default.
 
-To build a simple C file, something like this works:
+```bash
+./build.sh
 ```
-~/cross/bin/aarch64-w64-mingw32-gcc hello.c
-```
-I was able to cross compile zlib and install as follows:
+
+To skip dependencies bootstrapping process and fetching source code repositories for consecutive builds you can use environment variables:
+
+```bash
+RUN_BOOTSTRAP=0 UPDATE_SOURCES=0 ./build.sh
 ```
-make -f win32/Makefile.gcc DESTDIR=~/cross CC=~/cross/bin/aarch64-w64-mingw32-gcc AR=~/cross/bin/aarch64-w64-mingw32-ar RANLIB=~/cross/bin/aarch64-w64-mingw32-ranlib RC=~/cross/bin/aarch64-w64-mingw32-windres STRIP=~/cross/bin/aarch64-w64-mingw32-strip BINARY_PATH=$PREFIXDIR/bin INCLUDE_PATH=$PREFIXDIR/include LIBRARY_PATH=$PREFIXDIR/lib SHARED_MODE=1 PREFIX=aarch64-w64-mingw32- clean install
+
+The build script is doing multiple things:
+
+- Installs the dependency Ubuntu packages.
+- Clones several dependency source code repositories:
+  - modified binutils from [Windows-on-ARM-Experiments/binutils-woarm64](https://github.com/Windows-on-ARM-Experiments/binutils-woarm64),
+  - modified GCC from [Windows-on-ARM-Experiments/gcc-woarm64](https://github.com/Windows-on-ARM-Experiments/gcc-woarm64),
+  - modified MinGW from [Windows-on-ARM-Experiments/mingw-woarm64](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64).
+- Bootstraps the binutils and GCC sources with `gmp`, `mpfr`, `isl`, etc. dependencies.
+- Builds and installs binutils, stage 1 GCC, MinGW runtimes, GCC with libgcc, and then the entire MinGW.
+
+# Using the Cross-Compiler
+
+After building the toochain, to build a simple C source code file run:
+
+```bash
```

Comment:
I would add enablement of the toolchain in PATH and using aarch64-w64-mingw32-gc without full path.

Reply:
I would be also nice to describe how to build simple internal tests and run them.

Reply:
Changed to adding toolchain to `PATH`, added information on building the tests.